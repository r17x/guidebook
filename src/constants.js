const {
  NODE_ENV: env,
  REACT_APP_NAME: appName,
  REACT_APP_DESCRIPTION: appDescription,
} = process.env

export default {
    isDevelopment: env === 'development',
    isProduction: env === 'production',
    appName,
    appDescription,
}
