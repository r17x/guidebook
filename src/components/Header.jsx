import React from "react";
import { string } from "prop-types";
import { Div, Text } from "atomize";
import constants from "constants.js";

export default function Header({ title, onClick }) {
  return (
    <Div
      w="100%"
      p={{ y: "1rem" }}
      shadow="3"
      bg="white"
      textAlign="center"
      m={{ b: "1.5rem" }}
    >
      <Text
        tag="h1"
        data-testid="logo"
        onClick={onClick}
        textColor="black"
        textSize="display1"
      >
        {title}
      </Text>
    </Div>
  );
}

Header.propTypes = {
  title: string
};

Header.defaultProps = {
  title: constants.appName
};
