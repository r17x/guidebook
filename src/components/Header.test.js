import React from 'react'
import Header from 'components/Header'
import constants from 'constants.js'

test('should render header correctly', () => {
  const { container, getByText } = render(<Header />)
  const title = getByText(constants.appName)
  expect(title).toBeVisible()
  expect(container).toMatchSnapshot()
})
