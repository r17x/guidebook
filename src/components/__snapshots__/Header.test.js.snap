// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<div>
  <div
    class="sc-bdVaJa jJlxnC"
  >
    <h1
      class="sc-iwsKbI eQqwNh"
      data-testid="logo"
    >
      Guidebook
    </h1>
  </div>
</div>
`;
